src/librustc_const_eval/_match.rs

```diff
@@ -596,6 +602,9 @@ pub fn is_useful<'p, 'a: 'p, 'tcx: 'a>(cx: &mut MatchCheckCtxt<'a, 'tcx>,
     assert!(rows.iter().all(|r| r.len() == v.len()));
 
     let pcx = PatternContext {
+        // () is used to represent an unknown type in this context. If
+        // one of the fields has a known type, use it instead (other
+        // than that, all types should be equal modulo normalization).
```

Comment:
I don't really understand this comment. What is an unknown type here? And where is `()` being used?